base/shell/explorer/traywnd.cpp

```diff
@@ -805,6 +843,8 @@ class CTrayWindow :
             break;
 
         case ID_SHELL_CMD_UNDO_ACTION:
+            g_Arrangement = NONE;
+            RestoreWindowPos();
```

Comment:
It is interesting to observe that `RestoreWindowPos()` could itself set `g_Arrangement`to `NONE`...